coverage_point_calculator/src/lib.rs

```diff
@@ -0,0 +1,933 @@
+    WifiLocationScoreBelowThreshold(Decimal),
+    RadioThresholdNotMet,
+}
+
+impl BoostedHexStatus {
+    fn new(
+        radio_type: &RadioType,
+        location_trust_score: Decimal,
+        radio_threshold: &RadioThreshold,
+    ) -> Self {
+        // hip93: if radio is wifi & location_trust score multiplier < 0.75, no boosting
+        if radio_type.is_wifi() && location_trust_score < dec!(0.75) {
+            return Self::WifiLocationScoreBelowThreshold(location_trust_score);
+        }
+
+        // hip84: if radio has not met minimum data and subscriber thresholds, no boosting
+        if !radio_threshold.threshold_met() {
+            return Self::RadioThresholdNotMet;
+        }
+
+        Self::Eligible
+    }
+
+    fn is_eligible(&self) -> bool {
+        matches!(self, Self::Eligible)
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum RadioType {
+    IndoorWifi,
+    OutdoorWifi,
+    IndoorCbrs,
+    OutdoorCbrs,
+}
+
+impl RadioType {
+    fn base_coverage_points(&self, signal_level: &SignalLevel) -> Result<Decimal> {
+        let mult = match self {
+            RadioType::IndoorWifi => match signal_level {
+                SignalLevel::High => dec!(400),
+                SignalLevel::Low => dec!(100),
+                other => return Err(Error::InvalidSignalLevel(*other, *self)),
+            },
+            RadioType::OutdoorWifi => match signal_level {
+                SignalLevel::High => dec!(16),
+                SignalLevel::Medium => dec!(8),
+                SignalLevel::Low => dec!(4),
+                SignalLevel::None => dec!(0),
+            },
+            RadioType::IndoorCbrs => match signal_level {
+                SignalLevel::High => dec!(100),
+                SignalLevel::Low => dec!(25),
+                other => return Err(Error::InvalidSignalLevel(*other, *self)),
+            },
+            RadioType::OutdoorCbrs => match signal_level {
+                SignalLevel::High => dec!(4),
+                SignalLevel::Medium => dec!(2),
+                SignalLevel::Low => dec!(1),
+                SignalLevel::None => dec!(0),
+            },
+        };
+        Ok(mult)
+    }
+
+    fn rank_multipliers(&self) -> Vec<Decimal> {
+        match self {
+            RadioType::IndoorWifi => vec![dec!(1)],
+            RadioType::IndoorCbrs => vec![dec!(1)],
+            RadioType::OutdoorWifi => vec![dec!(1), dec!(0.5), dec!(0.25)],
+            RadioType::OutdoorCbrs => vec![dec!(1), dec!(0.5), dec!(0.25)],
+        }
+    }
+
+    pub fn is_wifi(&self) -> bool {
+        matches!(self, Self::IndoorWifi | Self::OutdoorWifi)
+    }
+
+    pub fn is_cbrs(&self) -> bool {
+        matches!(self, Self::IndoorCbrs | Self::OutdoorCbrs)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum RadioThreshold {
+    Verified,
+    UnVerified,
+}
+
+impl RadioThreshold {
+    fn threshold_met(&self) -> bool {
+        matches!(self, Self::Verified)
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
A lot of the tests in this module are asserting against hardcoded coverage point values rather than relative to the base_coverage_points for the radio type.  Feels like we could write these assertions so changes to the base_coverage_points of a radio_type would not cause these to fail. Could also make assertions harder to read/understand, so maybe not worth it?

Reply:
I've gone through and tried to clean most of the tests of concrete values. 
There are still a few (`boosted_hex`, and `location_trust_score_multiplier`) where I think the comment suffices for understanding the value being asserted against.